File: lib/utils/envValidation.ts
type EnvVar = string | undefined;

export function validateEnv(...requiredEnvVars: string[]): void {
  const missingVars = requiredEnvVars.filter(
    (varName) => !process.env[varName]
  );

  if (missingVars.length > 0) {
    throw new Error(
      `Missing required environment variables: ${missingVars.join(", ")}`
    );
  }
}

export function getEnvVar(name: string): string {
  const value = process.env[name] as EnvVar;
  if (!value) {
    throw new Error(`Environment variable ${name} is not set`);
  }
  return value;
}

File: bot/utils/env_validation.py
import os
from typing import List

def validate_env(*required_env_vars: str) -> None:
    missing_vars = [var for var in required_env_vars if not os.getenv(var)]
    if missing_vars:
        raise EnvironmentError(f"Missing required environment variables: {', '.join(missing_vars)}")

def get_env_var(name: str) -> str:
    value = os.getenv(name)
    if not value:
        raise EnvironmentError(f"Environment variable {name} is not set")
    return value